src/http/conf.rs

```diff
@@ -1,63 +1,267 @@
+        let conf_ctx = self.ctx.cast::<ngx_http_conf_ctx_t>();
+        let conf_ctx = conf_ctx.as_ref()?;
+        (*conf_ctx.loc_conf.add(module.ctx_index)).cast::<T>().as_mut()
+    }
+}
+
+impl HttpModuleConfExt for ngx_http_upstream_srv_conf_t {
+    unsafe fn http_server_conf_unchecked<T>(&self, module: &ngx_module_t) -> Option<&'static T> {
+        let conf = self.srv_conf;
+        if conf.is_null() {
+            return None;
+        }
+        (*conf.add(module.ctx_index)).cast::<T>().as_ref()
+    }
+    unsafe fn http_server_conf_mut_unchecked<T>(&self, module: &ngx_module_t) -> Option<&'static mut T> {
+        let conf = self.srv_conf;
+        if conf.is_null() {
+            return None;
+        }
+        (*conf.add(module.ctx_index)).cast::<T>().as_mut()
+    }
+}
+
+/// Trait to define and access main module configuration
+pub trait HttpModuleMainConf: HttpModule {
+    /// Type for main module configuration
+    type MainConf;
+    /// Get reference to main module configuration
+    fn main_conf(o: &impl HttpModuleConfExt) -> Option<&'static Self::MainConf> {
+        unsafe { o.http_main_conf_unchecked(Self::module()) }
+    }
+    /// Get mutable reference to main module configuration
+    fn main_conf_mut(o: &impl HttpModuleConfExt) -> Option<&'static mut Self::MainConf> {
+        unsafe { o.http_main_conf_mut_unchecked(Self::module()) }
+    }
+}
+
+/// Trait to define and access server-specific module configuration
+pub trait HttpModuleServerConf: HttpModule {
+    /// Type for server-specific module configuration
+    type ServerConf;
+    /// Get reference to server-level module configuration
+    fn server_conf(o: &impl HttpModuleConfExt) -> Option<&'static Self::ServerConf> {
+        unsafe { o.http_server_conf_unchecked(Self::module()) }
+    }
+    /// Get mutable reference to server-specific module configuration
+    fn server_conf_mut(o: &impl HttpModuleConfExt) -> Option<&'static mut Self::ServerConf> {
+        unsafe { o.http_server_conf_mut_unchecked(Self::module()) }
+    }
+}
+
+/// Trait to define and access location-specific module configuration
+pub trait HttpModuleLocationConf: HttpModule {
+    /// Type for location-specific module configuration
+    type LocationConf;
+    /// Get reference to location-specific module configuration
+    fn location_conf(o: &impl HttpModuleConfExt) -> Option<&'static Self::LocationConf> {
+        unsafe { o.http_location_conf_unchecked(Self::module()) }
+    }
+    /// Get mutable reference to location-level module configuration
+    fn location_conf_mut(o: &impl HttpModuleConfExt) -> Option<&'static mut Self::LocationConf> {
+        unsafe { o.http_location_conf_mut_unchecked(Self::module()) }
+    }
+}
+
+mod core {
+    use crate::ffi::{
+        ngx_http_core_loc_conf_t, ngx_http_core_main_conf_t, ngx_http_core_module, ngx_http_core_srv_conf_t,
+    };
+
+    /// Auxiliary structure to access core module configuration
```

Comment:
```suggestion
    /// Auxiliary structure to access `ngx_http_core_module` configuration.
```
(and similar changes for all the modules below).